pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -3111,6 +3111,15 @@ func (k kafkaManager) start(ctx context.Context, service string, envVars ...stri
 	// This isn't necessary for the nightly tests, but it's nice for iteration.
 	k.c.Run(ctx, option.WithNodes(k.kafkaSinkNodes), k.makeCommand("confluent", "local destroy || true"))
 	k.restart(ctx, service, envVars...)
+	// Wait for kafka to be ready. Otherwise we can sometimes try to connect to
+	// it too fast which fails the test.
+	k.waitForKafkaAvailable(ctx)
+}
+
+func (k kafkaManager) waitForKafkaAvailable(ctx context.Context) {
+	k.t.Status("waiting for kafka to be ready")
+	k.c.Run(ctx, option.WithNodes(k.kafkaSinkNodes), filepath.Join(k.binDir(), "kafka-topics"),
+		"--bootstrap-server=localhost:9092", "--list")
```

Comment:
Do we need to retry if it fails on the first attempt?

Reply:
no, this command should do retries itself for about 30 seconds.